[bug] Intel compilers: CMake does not find compiler after conan install

### Describe the bug

Dear Conan support, I am encountering the following issue:

### Environment
- ubuntu22.04 running on WSL2 (Windows 11).
- Conan  version: 2.19.1
- intel oneapi 2024.0 (base kit which includes compilers)

### Problem description
When I call 
```bash
conan install . -pr=<my-profile-using-intel-oneapi>
```
I'm getting the following error output:
```
[proc] Executing command: /usr/bin/cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_TOOLCHAIN_FILE=generators/conan_toolchain.cmake -S /build/src/test -B /build/src/test/build/RelWithDebInfo -G "Unix Makefiles"
[cmake] -- Using Conan toolchain: /build/src/test/build/RelWithDebInfo/generators/conan_toolchain.cmake
[cmake] -- Conan toolchain: Defining architecture flag: -m64
[cmake] -- Conan toolchain: Defining libcxx as C++ flags: -stdlib=libstdc++
[cmake] -- Conan toolchain: C++ Standard 20 with extensions ON
[cmake] CMake Error at /usr/share/cmake-3.28/Modules/CMakeDetermineCCompiler.cmake:49 (message):
[cmake]   Could not find compiler set in environment variable CC:
[cmake] 
[cmake]   icx.
[cmake] Call Stack (most recent call first):
[cmake]   CMakeLists.txt:3 (project)
[cmake] 
[cmake] 
[cmake] CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
[cmake] CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
```

When instead I run
```bash
conan build . -pr=<my-profile-using-intel-oneapi>
```
Then I'm able to subsequently use CMake directly with the conan-generated presets.

I guess the variables provided by `setvars.sh` script by Intel (especially the `PATH`) are not made persistent by Conan in the presets, hence the failure of CMake to actually find the compiler.

Thanks for your help!
- Milan

### How to reproduce it

Try to build a project using intel oneapi, by:
- using `conan install` to get the dependencies and CMake presets up
- then invoking cmake with the Conan-generated preset

--- a/conan/tools/intel/intel_cc.py
+++ b/conan/tools/intel/intel_cc.py
@@ -1,5 +1,3 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
 """
     Intel generator module for oneAPI Toolkits.
 
@@ -75,6 +73,10 @@ class IntelCC:
     def generate(self, scope="build"):
         """Generate the Conan Intel file to be loaded in build environment by default"""
         check_duplicated_generator(self, self._conanfile)
+        intel_cc_path = self._conanfile.conf.get("tools.intel:installation_path")
+        if intel_cc_path == "":
+            return
+
         if platform.system() == "Windows" and not self._conanfile.win_bash:
             content = textwrap.dedent("""\
                 @echo off

--- a/test/integration/toolchains/intel/test_intel_cc.py
+++ b/test/integration/toolchains/intel/test_intel_cc.py
@@ -1,3 +1,4 @@
+import os
 import platform
 import textwrap
 
@@ -55,3 +56,16 @@ def test_intel_cc_generator_linux():
     conanintelsetvars = get_intel_cc_generator_file(os_, installation_path, "conanintelsetvars.sh")
     expected = '. "/opt/intel/oneapi/setvars.sh" intel64'
     assert conanintelsetvars == expected
+
+
+def test_avoid_generation():
+    # The empty string for tools.intel:installation_path avoids the generation
+    profile = intelprofile % ("Linux", "")
+    client = TestClient()
+    client.save({
+        "conanfile.txt": conanfile,
+        "intelprofile": profile,
+    })
+    client.run("install . -pr intelprofile")
+    files = os.listdir(client.current_folder)
+    assert "conanintelsetvars" not in ",".join(files)

